### PR TITLE
[BD-5] [TNL-7282] [BB-2621] Support flexible peer grading

### PR DESCRIPTION
**Pull request by lullis**
This PR introduces support for flexible grading, i.e, a student can
receive their score to an ORA submission before getting the minimum
required peer grades, if at least floor(70% of defined minimum peer
grades are given)

**JIRA tickets**:
 - https://tasks.opencraft.com/browse/BB-2621
 - https://openedx.atlassian.net/browse/TNL-7282

Screenshots: Added "Enable Peer Flexible Grading" setting

**What changed?**

 - Added option to peer assessment grading options to enable/disable flexible grading
 - Added `enable_flexible_grading` attribute to peer assessment mixin
 - Added `_required_peer_grades` function to `openassessment/assessment/api/peer.py` determine if student can get submission score

**Reviewers**
- [ ] @xitij2000

### CODE REVIEW

**Comment by xitij2000** [comment]
@lullis Could you rebase this PR on the current master? It's currently including a lot of other commits making it hard to review.

File: openassessment/assessment/api/peer.py [hunk 1]
```diff
@@ -168,7 +179,7 @@ def get_score(submission_uuid, peer_requirements):
         assessment__score_type=PEER_TYPE
     ).order_by('-assessment')
 
-    submission_finished = items.count() >= peer_requirements["must_be_graded_by"]
+    submission_finished = items.count() >= _required_peer_grades(workflow, peer_requirements)
```
Comment (xitij2000) [inline openassessment/assessment/api/peer.py]: I think it might make sense to move this entire evaluation into the function, i.e. call the function: `_check_submission_finished` or `_is_submission_finished` etc..

File: openassessment/templates/openassessmentblock/edit/oa_edit_peer_assessment.html [hunk 2]
```diff
@@ -26,13 +26,29 @@
                 </li>
                 <li class="field comp-setting-entry">
                     <div class="wrapper-comp-setting">
-                        <label for="peer_assessment_graded_by" class="setting-label"> {% trans "Graded By" %}</label>
+                        <label for="peer_assessment_graded_by" class="setting-label">{% trans "Graded By" %}</label>
```
Comment (xitij2000) [inline openassessment/templates/openassessmentblock/edit/oa_edit_peer_assessment.html]: nit: avoid unnecessary diff.

File: openassessment/templates/openassessmentblock/edit/oa_edit_peer_assessment.html [hunk 3]
```diff
@@ -26,13 +26,29 @@
...
+                        {% trans "True"%}
+                      </option>
+                    </select>
+                  </div>
+                  <p class="setting-help">
+                    {% trans "Learners who have received a minimum of {floor-30% of required peer-reviews configuration value} who have waited longer than 7 days will be given their grade instead of having to wait longer for more peer reviews"%}
+                  </p>
+                </li>
+                <li class="field comp-setting-entry">
+                  <div class="wrapper-comp-setting">
```
Comment (xitij2000) [inline openassessment/templates/openassessmentblock/edit/oa_edit_peer_assessment.html]: nit: doesn't seem to be matching the code style of the rest of the file.

File: openassessment/xblock/schema.py [hunk 4]
```diff
@@ -121,6 +121,7 @@ def datetime_validator(value):
             'required': bool,
             'must_grade': All(int, Range(min=0)),
             'must_be_graded_by': All(int, Range(min=0)),
+            Required('enable_flexible_grading', default=True): bool,
```
Comment (xitij2000) [inline openassessment/xblock/schema.py]: It should default to false, since that is the current behaviour.

File: openassessment/xblock/static/js/src/studio/oa_edit_assessment.js [hunk 5]
```diff
@@ -57,6 +57,7 @@ OpenAssessment.EditPeerAssessmentView.prototype = {
      {
          must_grade: 5,
          must_be_graded_by: 2,
+         enable_flexible_grading: true,
```
Comment (xitij2000) [inline openassessment/xblock/static/js/src/studio/oa_edit_assessment.js:70]: nit: default to false.

**Review by xitij2000 — COMMENTED** [review]
This is just a first pass of comments. I will test properly and review later.

File: openassessment/xblock/schema.py (hunk 4 above)
Comment (lullis) [reply openassessment/xblock/schema.py]: I'd agree with you and keep the existing behavior, but the ticket description says "By default for problems we’d like to have this be enabled but for existing problems this might be disabled.", which led me to switch to true.

Comment (xitij2000) [reply openassessment/xblock/schema.py]: Alright, in that case, we need to make sure that any existing blocks that don't have this field are handled properly, i.e. if this field is undefined, it should default to false. 

So I think it can default to false in the schema, but default to true for the data used to fill in the default values in `defaults.py`.
